### ISSUE TITLE
Using VK_DEPENDENCY_BY_REGION_BIT when rendering to swapchain should generate a warning

### ISSUE THREAD

**Issue by hrydgard** [KhronosGroup/Vulkan-ValidationLayers#1093]
I accidentally had VK_DEPENDENCY_BY_REGION_BIT specified in my subpass dependency for a render pass rendering to a swapchain image:

```
	VkSubpassDependency dep{};
	dep.srcSubpass = VK_SUBPASS_EXTERNAL;
	dep.dstSubpass = 0;
	dep.srcStageMask = VK_PIPELINE_STAGE_COLOR_ATTACHMENT_OUTPUT_BIT;
	dep.srcAccessMask = 0;
	dep.dstStageMask = VK_PIPELINE_STAGE_COLOR_ATTACHMENT_OUTPUT_BIT;
	dep.dstAccessMask = VK_ACCESS_COLOR_ATTACHMENT_READ_BIT | VK_ACCESS_COLOR_ATTACHMENT_WRITE_BIT;
	dep.dependencyFlags = VK_DEPENDENCY_BY_REGION_BIT;
```

This causes ugly tearing on Mali. I think the validation layer should at least warn about this, since it doesn't make much sense to use a BY_REGION dependency here.

**Comment by hrydgard** [comment]
Actually the tearing was really caused by my pWaitDstStageMask array being out of scope at the time of vkQueueSubmit. So I'm not sure if this is of any importance...

**Comment by mark-lunarg** [comment]
Thanks @hrydgard -- we typically don't put warnings in the Khronos Layer, but we do have a place for them in the Assistant Layer ( #24 ).  If you feel that this would still be a useful warning just add a link to this issue there and we'll add it to the list.
